src/game.cpp

```diff
@@ -10153,40 +10170,58 @@ bool game::plfire_check( item &weapon, int &reload_time ) {
     return okay;
 }
 
-bool game::plfire( item *weapon, int bp_cost, bool held )
+bool game::plfire( item *weapon, int bp_cost, targeting_data *tdata )
 {
     static int bio_power_cost = 0;
-    static item *cached_weapon = nullptr;
+    static item *cached_weapon = &u.weapon;
     static bool held_weapon = true;
-    
-    if( weapon != nullptr ) {
-        // valid weapon, set the cached weapon and bp_cost to the current values.
-        cached_weapon = weapon;
+    static targeting_data args;
```

Comment:
I'd go with `player` since a `game` might interact with more than one `player` trying to fire. A `player` probably wouldn't deal with more than one `game` at a time.